pp.c

```diff
@@ -817,18 +817,11 @@ S_do_chomp(pTHX_ SV *retval, SV *sv, bool chomping)
                     /* Assumption is that rs is shorter than the scalar.  */
                     if (SvUTF8(PL_rs)) {
                         /* RS is utf8, scalar is 8 bit.  */
-                        bool is_utf8 = TRUE;
-                        temp_buffer = (char*)bytes_from_utf8((U8*)rsptr,
-                                                             &rslen, &is_utf8);
-                        if (is_utf8) {
-                            /* Cannot downgrade, therefore cannot possibly match.
-                               At this point, temp_buffer is not alloced, and
-                               is the buffer inside PL_rs, so don't free it.
-                             */
-                            assert (temp_buffer == rsptr);
+                        if (! utf8_to_bytes_temp_pv(&rsptr, &rslen)) {
+                            /* Cannot downgrade, therefore cannot possibly
+                             * match. */
```

Comment:
do_chomp() can be called for many, many strings (think `chomp(my @x = <$fh>)` at a time, by using the temp_pv() variant here we're possibly retaining many, many temp strings to be freed later, and expanding the save stack by many, many entries.

I think this should use the new_pv() variant.

Reply:
Good point; done